concourse/client/api.py

```diff
@@ -221,9 +221,24 @@ def build_events(self, build_id):
     @ensure_annotations
     def trigger_resource_check(self, pipeline_name: str, resource_name: str):
         url = self.routes.resource_check(pipeline_name=pipeline_name, resource_name=resource_name)
-        # Resource checks are triggered by a POST with an empty JSON-document as body against
-        # the resource's check-url
-        self._post(url, body='{}')
+        max_retries = 5
+
+        def _try_resource_check(current_attempt):
+            # Resource checks are triggered by a POST with an empty JSON-document as body against
+            # the resource's check-url
+            response = self.request_builder.post(url, check_http_code=False, body='{}', headers={})
+            if response.status_code == 500:
+                if response.reason.startswith('parent type has no version'):
+                    if current_attempt <= max_retries:
+                        return _try_resource_check(current_attempt+1)
+                    else:
+                        raise RuntimeError(
+                            f"Unable to check resource '{resource_name}' in pipeline "
+                            f"'{pipeline_name}'. If this pipeline was recently deployed, please "
+                            "try again in a minute."
+                        )
+            else:
```

Comment:
i think this else could be removed

how about determining the error-type (500 + reason-str), and store it (e.g. `is_pending_initialisation = True`, or similar).

then either early-exit for a retry, or raise (either with special, or generic error-msg)? should be doable w/o nesting deeper than two